src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx

```diff
@@ -100,6 +119,68 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
     }
   })
 
+  const validateServiceReconcile = async (): Promise<void> => {
+    try {
+      const response = await validateTemplateInputsPromise({
+        queryParams: {
+          accountIdentifier: accountId,
+          orgIdentifier,
+          projectIdentifier,
+          identifier: serviceId
+        }
+      })
+      clear()
+      if (response?.data?.validYaml === false && response?.data?.errorNodeSummary) {
+        setValidateTemplateInputsResponse(response)
+        setShouldShowOutOfSyncError(true)
+      } else {
+        showSuccess(getString('pipeline.outOfSyncErrorStrip.noErrorText', { entity: TemplateErrorEntity.SERVICE }))
+        setShouldShowOutOfSyncError(false)
+      }
+    } catch (reconcileValidateError) {
+      if (reconcileValidateError && shouldShowError(reconcileValidateError)) {
+        showError(getErrorInfoFromErrorObject(reconcileValidateError))
+      }
+    }
+  }
+
+  const handleReconcileClick = () => {
+    validateServiceReconcile()
+    showSuccess(getString('pipeline.outOfSyncErrorStrip.reconcileStarted'))
+  }
+
+  const updateAndPublishReconciledService = async (reconciledServiceYaml: string): Promise<void> => {
+    clear()
+    const reconciledService = parse(reconciledServiceYaml) as NGServiceConfig
+    const body = {
+      ...omit(cloneDeep(reconciledService?.service), 'serviceDefinition', 'gitOpsEnabled'),
+      projectIdentifier: isServiceCreateModalView ? projectIdentifier : serviceData?.projectIdentifier,
+      orgIdentifier: isServiceCreateModalView ? orgIdentifier : serviceData?.orgIdentifier,
+      //serviceId is not present in queryParam when service is created in pipeline studio.
+      identifier: defaultTo(serviceId, reconciledService?.service?.identifier),
+      yaml: reconciledServiceYaml
+    }
+
+    try {
+      const response = await updateService(body)
+      if (response.status === 'SUCCESS') {
+        // We invalidate the service inputs call on updating an existing service
+        queryClient.invalidateQueries(['getServicesYamlAndRuntimeInputs'])
+
+        showSuccess(getString('common.serviceUpdated'))
+        setShouldShowOutOfSyncError(false)
+        fetchPipeline({ forceFetch: true, forceUpdate: true })
+        setIsDeploymentTypeDisabled?.(!!reconciledService.service?.serviceDefinition?.type)
+      } else {
+        throw response
+      }
+    } catch (e: any) {
+      showError(e?.data?.message || e?.message || getString('commonError'))
```

Comment:
```suggestion
      showError(getErrorInfoFromErrorObject(e))
```
Please check if this satisfy the requirement

Reply:
done